rest_framework/relations.py

```diff
@@ -492,10 +492,19 @@ def to_internal_value(self, data):
         if not self.allow_empty and len(data) == 0:
             self.fail('empty')
 
-        return [
-            self.child_relation.to_internal_value(item)
-            for item in data
-        ]
+        if isinstance(self.child_relation, PrimaryKeyRelatedField):
+            values = list(self.child_relation.get_queryset().filter(pk__in=data))
```

Comment:
Why not use `values_list('pk', flat=True)` here to just fetch the ids and skip object instantiation?